### PR TITLE
MM-35035: Cypress parallelism

### PR DESCRIPTION
**Pull request by Phrynobatrachus**
#### Summary
This enables parallel test runs in CircleCI. Based on some test runs it cuts the Cypress jobs to ~9 minutes, which means the whole workflow will also be ~9 min since those are the longest running jobs.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-35035

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -140,9 +140,10 @@ commands:
           name: Run Cypress Tests
           no_output_timeout: 30m
           command: |
-            export FAILURE_MESSAGE="At least one test has failed."
-            export RESULTS_OUTPUT="results-output.txt"
-            cd tests-e2e && npm run test |& tee $RESULTS_OUTPUT; if grep "$FAILURE_MESSAGE" "$RESULTS_OUTPUT"; then exit 1; fi
+            cd tests-e2e
+            circleci tests glob "cypress/integration/**/*.js" | circleci tests split --split-by=timings > ./tests-to-run
+            npm run test -- --spec "$(cat ./tests-to-run | tr '\n' ',')"
```
Comment (lieut-data) [inline .circleci/config.yml:145]: Any idea what the `results-output.txt` used to do, @cpoile?

**Review by lieut-data — APPROVED** [review]
Crazy that 6 lines of code will save us hours and hours of waiting....

File: .circleci/config.yml (hunk 1 above)
Comment (Phrynobatrachus) [reply .circleci/config.yml:145]: Oh yeah meant to note this removal, I couldn't find anywhere that output file was used and the job ends with `exit 1` on failure even without it. Definitely can put it back if there's something I missed with it though.

Comment (cpoile) [reply .circleci/config.yml:145]: ...Nope. :)

**Review by cpoile — APPROVED** [review]
Great # lines to performance improvement ratio, @Phrynobatrachus!

**Pull request merged**